### PR TITLE
Smooth scrolling and overlap issue fix

### PR DESCRIPTION
**Pull request by merwane**
## Description

- Added a native solution to handle smooth scrolling
- Added a native solution to fix the overlapping issue

## Related Issue
#292 
#294 

## Screen recording:

https://drive.google.com/file/d/1LeEoUjXu82WxNbtRn2QkhCyqOlhMGN8r/view?usp=sharing

## Checklist:

- [x] I have read the contributing guidelines in the project README.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by merwane** [comment]
Done :)

https://drive.google.com/file/d/1lJuqXerpZs77E18hBHXnMXP79lnS8vst/view?usp=sharing

File: docs/.vuepress/theme/styles/theme.styl [hunk 1]
```diff
@@ -36,7 +37,8 @@ h1
 h2
   font-size $fsLarge
   font-weight 400
-  margin-top 2em
+  padding-top 4em
```
Comment (merwane) [inline docs/.vuepress/theme/styles/theme.styl]: native fix to prevent overlap

**Comment by samajammin** [comment]
Nice! The only issue I'm seeing now is that after the smooth scroll, the sidebar menu is not highlighting the correct section. Example below - when I select "Block Explorers" section, the sidebar highlights "Testing Tools":

Screen Recording 2019-10-09 at 05 00 PM

Any idea on how to solve this? One way might be configuring the plugin we use:
https://vuepress.vuejs.org/plugin/official/plugin-active-header-links.html#install

**Comment by merwane** [comment]
Strange, maybe it has something to do with the resized margins? I'll try to figure it out

**Comment by merwane** [comment]
It appears Vuepress's sidebar has a small conflict when smooth scrolling is set. However, Vuepress should release a smooth scrolling feature soon, this will probably fix the issue. 
https://github.com/vuejs/vuepress/issues/1930

**Comment by merwane** [comment]
Am I eligible for the [bounty](https://gitcoin.co/issue/ethereum/ethereum-org-website/294/3534) on Gitcoin after this pull request? :)

**Pull request closed without merging**